src/palindrom.js

```diff
@@ -193,319 +187,317 @@ var Palindrom = (function() {
+      establish(this, this.remoteUrl.href, null, bootstrap);
+    }
 
-  // TODO: auto-configure here #38 (tomalec)
-  function establish(network, url, body, bootstrap) {
-    return network.xhr(url, 'application/json', body, function(res) {
-      bootstrap(res.data);
-      if (network.useWebSocket) {
-        network.webSocketUpgrade(network.onSocketOpened);
+    reestablish(pending, bootstrap) {
+      establish(
+        this,
+        `${this.remoteUrl.href}/reconnect`,
+        JSON.stringify(pending),
+        bootstrap
+      );
+    }
+
+    /**
+     * Send any text message by currently established channel
+     * @TODO: handle readyState 2-CLOSING & 3-CLOSED (tomalec)
+     * @param  {String} msg message to be sent
+     * @return {PalindromNetworkChannel}     self
+     */
+    send(msg) {
+      // send message only if there is a working ws connection
+      if (this.useWebSocket && this._ws && this._ws.readyState === 1) {
+        this._ws.send(msg);
+        this.onSend(msg, this._ws.url, 'WS');
+      } else {
+        const url = this.remoteUrl.href;
+        this.xhr(url, 'application/json-patch+json', msg, (res, method) => {
+          this.onReceive(res.data, url, method);
+        });
       }
-    });
-  }
-  /**
-   * Send any text message by currently established channel
-   * @TODO: handle readyState 2-CLOSING & 3-CLOSED (tomalec)
-   * @param  {String} msg message to be sent
-   * @return {PalindromNetworkChannel}     self
-   */
-  PalindromNetworkChannel.prototype.send = function(msg) {
-    var that = this;
-    // send message only if there is a working ws connection
-    if (this.useWebSocket && this._ws && this._ws.readyState === 1) {
-      this._ws.send(msg);
-      that.onSend(msg, that._ws.url, 'WS');
-    } else {
-      var url = this.remoteUrl.href;
-      this.xhr(url, 'application/json-patch+json', msg, function(res, method) {
-        that.onReceive(res.data, url, method);
-      });
+      return this;
     }
-    return this;
-  };
-  /**
-   * Callback function that will be called once message from remote comes.
-   * @param {String} [JSONPatch_sequences] message with Array of JSONPatches that were send by remote.
-   * @return {[type]} [description]
-   */
-  PalindromNetworkChannel.prototype.onReceive = function(/*String_with_JSONPatch_sequences*/) {};
-  PalindromNetworkChannel.prototype.onSend = function() {};
-  PalindromNetworkChannel.prototype.onStateChange = function() {};
-  PalindromNetworkChannel.prototype.upgrade = function(msg) {};
 
-  function closeWsIfNeeded(network) {
-    if (network._ws) {
-      network._ws.onclose = function() {};
-      network._ws.close();
-      network._ws = null;
+    /**
+     * Callback function that will be called once message from remote comes.
+     * @param {String} [JSONPatch_sequences] message with Array of JSONPatches that were send by remote.
+     * @return {[type]} [description]
+     */
+    onReceive() /*String_with_JSONPatch_sequences*/ {
```

Comment:
Shouldn't the type comment be inside `()`?

Reply:
Done.